torch/distributed/tensor/experimental/_attention.py

```diff
@@ -1291,3 +1291,22 @@ def context_parallel_unshard(
         else _SequentialSharder
     )
     return [sharder.unshard(b, mesh, dim) for b, dim in zip(buffers, seq_dims)]
+
+
+def set_rotate_method(rotate_method: str) -> None:
+    """
+    Context Parallel SDPA requires the rotation of kv shards. Users can call this
+    API to specify which rotation method to use. "alltoall" shuffles the kv shards
+    using all-to-all collective while "allgather" gathers the kv shards using
+    all-gather collective. If this API has not been called, the default rotate
```

Comment:
> while "allgather" gathers the kv shards using all-gather collective.

Can we change this to 

>. while "allgather" gathers the kv shards using all-gather collective after the first sub-SDPA computation.

We may want to implement the allgather version that happen before all sub-SDPA computation.